integration/store_gateway_limits_hit_test.go

```diff
@@ -42,6 +42,7 @@ func Test_MaxSeriesAndChunksPerQueryLimitHit(t *testing.T) {
 			"-blocks-storage.tsdb.retention-period":             blockRangePeriod.String(), // We want blocks to be immediately deleted from ingesters.
 			"-blocks-storage.tsdb.ship-interval":                "1s",
 			"-blocks-storage.tsdb.head-compaction-interval":     "500ms",
+			"-compactor.first-level-compaction-wait-period":     "1m", // Do not compact aggressively
```

Comment:
If this is the default you want for integration tests, then it's better to do the change in `BlocksStorageFlags` (defined in `integration/configs.go`).

Reply:
I don't think this is necessarily a better default since making this the default causes some other ITs, which expect immediate L1 compaction, to fail. I'll leave the current default for now and just override for these 2 tests.